### DISCUSSION TITLE
Getting the correct Nightscout Synce (Rest-API) URL for AndroidAAPS to work or xdrip

### DISCUSSION THREAD

**Discussion by Atlas-Night-Out** [NightscoutFoundation/xDrip#2355] [Q&A]
I'm really struggling to get a working URL to work from xdrip+ onto a Nightscout Web browers and to AndroidAPS. I have done all the settings to what you would think would work but are still now working on a google server.

These are some I have tried aand still trying!

https://yoursite.herokuapp.com/api/v1/
https://yoursite.up.railway.app/api/v1/
https://yoursite.10be.de/api/v1/
https://Token@hostname assuming Token is the token from the master, generated as explained above, and hostname is your noip.com hostname

This also does not work for a Google Server! 
https://API_SECRET@my_nightscout_url/api/v1/
my_nightscout_url will depend on where your site is hosted (herokuapp.com, railway.app, fly.io, ...)
You will find your API_SECRETin your site variables.

I can get Nightscout to work fine on a Google Server and get readings to it too. I just cannot get it to do what Heroku can do from withing xdrip.

I have also tried this and this also does not work!

(This is what you need to do in xDrip on your master phone.
Go to xDrip Settings −> Cloud Upload −> Nightscout Sync (REST-API).
Enable at the top.
Let’s say the hostname you chose when you signed up for noip.com was hostname. And let’s say you set the API secret, while finalizing the Nightscout installation as explained previously, to API-Secret.
Tap on Base URL. Enter the following.
https://API-Secret@hostname/api/v1/) 

I have put API-Secret@ noip hostname/api/v1      with no luck! 

Any help on trying to resolve this issues please.

I'm also not very good at fully explaining so if you are not sure what I mean please ask me!

**Comment by Navid200** [comment]
Can you log into your Nightscout on Google Cloud through a web browser?

**Reply by Navid200** [reply]
I have updated the guide (setup page) with more detail hoping to provide more clarity.
https://navid200.github.io/xDrip/docs/Nightscout/NS_setup

**Reply by Atlas-Night-Out** [reply]
yes that seems to do it auto. just gone back to heroku now and even thast not uploading agian seems very buggy with me!
No data seems to be uploading from xdrip or AAPS since all the changes again I had it fine again yeterday with heroku but not google! Now I'm back to both not working right!

**Reply by Atlas-Night-Out** [reply]
@Navid200 yes it makes sence it did last time but its not working! and what I mean by that is the data does not upload to nightscout.I get the readings! 5.0mmols but that is it. Then at times I can get some reading from xdrip none from AAPS then some from AASP and none from xdrip then I get errors in AASP for nightscout then I go back to Heroku and but my setting back to what they where and all works fine. Normaly but some time like now again they don't work and I have to play around for days to try to get it to all work again. eventualy it does but Its somthing that is very hard to pin point what is th ecause of it all. I know a few are spelling mistakes some are settings being wrong but when you check them and they seem correct you are at a loss again like I keep being!

**Reply by Atlas-Night-Out** [reply]
everything stopped working now so seems like nightscouts down yet again!

**Reply by Atlas-Night-Out** [reply]
@Navid200 just wondering why Google cloud as just stopped working from no where? no setting changed or anything now nothing?

**Reply by Navid200** [reply]
I don't know.

Going to your Google Cloud console dashboard and selecting compute engine, you can see if your virtual machine is active or not.

If it is, everything should be working fine.

**Reply by Atlas-Night-Out** [reply]
@Navid200   not sure now as done that much testing Im totaly lost on ot all. google cloud tends to take a while to remove old deleted accounts which is frustrating as I do not remeber the new ones name but what I have go tinto seems ok and running as a price tag at the side of it but that was all. No account running though and no heroku account running now both went down at same time! maybe they are down for a while no point trying to do anything until tomorrow now will only make things worse. Heroku is asking em to check log files so strange when nothign as changed!

**Reply by Atlas-Night-Out** [reply]
this is all im getting no nightscout working just stopped with no warning or anything! [
![google istances](https://user-images.githubusercontent.com/83882497/192731890-c6ecda73-89b5-4570-93af-0615d890a89f.jpg)
](url)

**Reply by Atlas-Night-Out** [reply]
@Navid200  no clue how I got it back on but all I did was turn the enbled off on it to see if it would refresh it and this seems to have got it back on! Now I need to find out how to get data too it.

**Reply by Atlas-Night-Out** [reply]
I can get data easily up to nightscout in xdrip this is not the issues here! But in AAPS the url that is added is not correct and I need to find the correct way to add it. There is an option in (Rest-API) in the Extra Options that does this! But that needs to be turned off if doing it from AAPS. Other wise when it does work you would get double data going into it. But for me at the moment I need to keep it on until AAPS can get data back to xdrip and nightscout! This is the error I get in AASP at the moment, I have tried other ways with the URL Code with no luck. [
![tempFileForShare_20220928-100432](https://user-images.githubusercontent.com/83882497/192743189-6da73c48-26e5-4f08-8619-96e046b4f03f.jpg)
](url)

**Reply by Atlas-Night-Out** [reply]
@Navid200  There seems to be a bug in AASP as I have just re - installed the software changed over to a Dev server and everything is working again carbs and insulin units now going up to nightscout as well as xdrip so seems like I have resolved my own issues! Thanks!
